pandas/tseries/period.py

```diff
@@ -388,10 +388,10 @@ def _array_values(self):
 
     def astype(self, dtype):
         dtype = np.dtype(dtype)
-        if dtype == np.object_:
-            return Index(np.array(list(self), dtype), dtype)
```

Comment:
I also changed this line because it seems that `self.asobject` (i.e. `DatetimeIndexOpsMixin.asobject`) does essentially the same thing.